### DISCUSSION TITLE
Commit verification with SSH signed commits possible?

### DISCUSSION THREAD

**Discussion by dakky** [fluxcd/flux2#4145] [General] [answered]
At the moment its possible ti let flux accept only signed commits (see https://fluxcd.io/flux/components/source/gitrepositories/#verification)

The examples are only mentioning GPG signed commits. Github and Gitlab also allow SSH signed commits. 
**Can flux also handle ssh pub keys in order to verify the commit?**

**Comment by makkes** [comment, marked as answer]
go-git, the upstream library we use in Flux, [doesn't support this](https://github.com/go-git/go-git/issues/400), yet, but as soon as that issue is addressed, we will start incorporating it into Flux itself.

**Reply by dakky** [reply]
Thanks: the upstream issue is https://github.com/go-git/go-git/issues/400

**Reply by hiddeco** [reply]
We theoretically have the means now to do manual verification outside of go-git. As I am currently having issues with finding time to finish https://github.com/go-git/go-git/pull/705, but did take the time to write https://github.com/hiddeco/sshsig.

Only downside is that there is no bit yet which understands an [allowed signers](https://man.openbsd.org/ssh-keygen.1#ALLOWED_SIGNERS) entry, which would mean we would have to continue the work in https://github.com/hiddeco/sshsig/pull/7 to at least be able to connect a principal (email address) to an allowed public key.

**Reply by hkrutzer** [reply]
That issue has been closed with https://github.com/go-git/go-git/issues/1849 as a followup, does this feature now depend on that followup issue?

**Reply by bb-Ricardo** [reply]
does this help?: https://github.com/fluxcd/pkg/pull/1141
